docs: blog - three Ways to Integrate Automatic SQL Review into GitHub (#688)
bytebase/bytebase.com: content/blog/integrate-sql-review-into-github.md

---
title: Three Ways to Integrate Automatic SQL Review into GitHub
author: Changyu
published_at: 2022/12/08 09:00
feature_image: /static/blog/integrate-sql-review-into-github/overview.webp
tags: Explanation
featured: true
description: The three ways to integrate Bytebase SQL Review with GitHub. GitHub Action, GitHub App, and GitOps CI
---

## Overview

The review of SQL change scripts is a critical part of the database change process, which used to be
the sole responsibility of the DBA. Developers often submitted the SQL scripts to the DBA just before

> going live, making it difficult to ensure the quality of changes.

Reviewer: ```suggestion
the changes go live, leaving the DBAs minimal time to review the scripts (or request changes), making it difficult to ensure the quality of changes.
```

✅ **Lightweight.** Users don't need to deploy any additional components, just one click to deploy the
GitHub App and activate it.

✅ **Easy to configure.** Point-and-Click configuration.

✅ **Comprehensive presentation.** You can view the SQL Review results in both the PR overview and
per-file page.

❌ **Basic SQL Review capability.** Due to the inability to connect to a database, this mode can only
review SQL text based on static rules.

❌ **Explicitly specify the database type.** Due to the inability to connect to a database, this schema
can only be explicitly configured to take effect on one of the supported database types.

❌ **Single active SQL Review policy.** Only one set of SQL Review rules are allowed to be active at
the same time.

❌ **CI-only.** This mode only covers SQL Review and is not integrated with the subsequent SQL
deployment processes.

## GitOps CI

The Bytebase Console-based integration mode provides more powerful functionality:

1. Unlock more advanced SQL Review rules by connecting with the changed database.
1. Integrate with the subsequent CD processes.

### Setup

1. Deploy Bytebase Console and complete the basic configuration

>    This mode requires the deployment of Bytebase Console and import the database enviornment into it.

Reviewer: ```suggestion
   This mode requires the deployment of Bytebase Console, and you need to import your database environment into Bytebase.
```

   Please refer to [Bytebase Installation Guide](https://www.bytebase.com/docs/get-started/install/overview)
   for the detailed process.

1. Activate SQL Review CI with one click
   The GitHub Action can be configured automatically by configuring GitOps Workflow in the Project
   Configuration page from Bytebase Console.

> 1. Customizing SQL Review rules

Reviewer: ```suggestion
1. Customizing SQL Review rules

```

>    If you are not satisfied with the default SQL Review rules, you can configure it visually in the

Reviewer: ```suggestion
   If you are not satisfied with the default SQL Review rules, you can configure it visually from the
```

configured, associated with specific environments (Test, Prod).

✅ **Integration with CD.** Automatically triggers the subsequent SQL deployment process after
the SQL Review passes and the code is merged, enabling a streamlined database CI/CD experience.
This unifies the SQL Review standards between the application developer and DBA teams.

✅ **No need to specify database type.** Bytebase can connect to the database to determine the database
type automatically.

❌ **Paid subscription.** Bytebase Enterprise subscription is required.

❌ **More setup.** Users need to deploy Bytebase Console and prepare the database environment.

❌ **Single presentation.** SQL Review results can only be viewed in the file being reviewed and
there is no overview of SQL Review results.

## Comparison Table

|                       | GitHub Action                 | GitHub App                                    | GitOps CI                                                          |
| --------------------- | ----------------------------- | --------------------------------------------- | ------------------------------------------------------------------ |

> | For                   | **Personal and small teams**  | **Personal and small teams**                  | **Teams needing complete database change management**              |

Reviewer: is it better to use "individual" here instead of personal?

Author: True

> | For                   | **Personal and small teams**  | **Personal and small teams**                  | **Teams needing complete database change management**              |

Author: ```suggestion
| For                   | **Individual and small teams**  | **Individual and small teams**                  | **Teams needing complete database change management**              |
```
